src/audio/google/google_rtc_audio_processing_mock.c

```diff
@@ -145,11 +147,17 @@ int GoogleRtcAudioProcessingProcessCapture_int16(GoogleRtcAudioProcessingState *
 	int16_t *ref = state->aec_reference;
 	int16_t *mic = (int16_t *) src;
 	int16_t *out = dest;
-	int n;
+	int channels;
+	int n, m;
+
+	channels = MIN(state->num_capture_channels, state->num_aec_reference_channels);
+	channels = MIN(state->num_output_channels, channels);
```

Comment:
@singalsu this works for the 2ch case where the PDM 0/1 are configured on the board but if only PDM 2/3 are connected, this won't work correctly right?

Reply:
in an ideal world, we would reduce our channels to only the ones we care about so we don't need these silly configs and only have real channels in the data.

Reply:
I changed it to version where every output channel has a mix of mic and reference. Hope it's good for testing now.